### PR TITLE
✨  Hot Swappable Field Configurations

### PR DESCRIPTION
**Pull request by jossehblanco**
This PR adds a utility function that allows the field's configuration to be changed and re-validated. It also adds some of the missing documentation comments.

### CODE REVIEW

File: Sources/CombineForms/Form/CombineFormValidating.swift [hunk 1]
```diff
@@ -10,17 +10,28 @@ import Foundation
 import SwiftUI
 
 public protocol CombineFormValidating: AnyObservableObject {
+    /// Property that determines if a form is valid or not
     var formValid: Bool { get set }
+    /// A formatted list of all the errors found in the form's fields
     var formErrors: String { get set }
+    /// Seprarator string for the form errors string. Format: <Field label> <Separator> <field error message>
```
Comment (RagdeSey) [inline Sources/CombineForms/Form/CombineFormValidating.swift]: A typo here  `Separator`

File: Sources/CombineForms/FormField/CombineFormField.swift [hunk 2]
```diff
@@ -43,18 +50,27 @@ public class CombineFormField: ValidatableField, ObservableObject, Hashable {
         }
     }
     
+    /// A sring that identifies the requirements for the field. Examples: Optional, Required, etc.
```
Comment (RagdeSey) [inline Sources/CombineForms/FormField/CombineFormField.swift]: Typo here `String`

File: Sources/CombineForms/FormField/CombineFormField.swift [hunk 3]
```diff
@@ -116,6 +132,14 @@ public class CombineFormField: ValidatableField, ObservableObject, Hashable {
         }
     }
     
+    /// Replaces the field's current configuration, and re-validates it according to its new rules
+    public func replaceCurrentConfiguration(with newConfiguration: CombineFormFieldConfiguration) {
+        cancellables.forEach { $0.cancel() }
+        cancellables.removeAll()
+        self.configuration = newConfiguration
```
Comment (RagdeSey) [inline Sources/CombineForms/FormField/CombineFormField.swift]: the self is unnecessary here :D

**Review by RagdeSey — CHANGES_REQUESTED** [review]

File: Sources/CombineForms/Form/CombineFormValidating.swift (hunk 1 above)
Comment (jossehblanco) [reply Sources/CombineForms/Form/CombineFormValidating.swift]: Thanks I didn't notice it 😂

File: Sources/CombineForms/FormField/CombineFormField.swift (hunk 2 above)
Comment (jossehblanco) [reply Sources/CombineForms/FormField/CombineFormField.swift]: Done 😅

File: Sources/CombineForms/FormField/CombineFormField.swift (hunk 3 above)
Comment (jossehblanco) [reply Sources/CombineForms/FormField/CombineFormField.swift]: You're right! I removed it

**Review by RagdeSey — APPROVED** [review]
Well done

**Pull request merged**